### PR TITLE
Added a sampling-based approach for retrieving counts

### PR DESCRIPTION
**Pull request by MichaelRoeder**

### CODE REVIEW

**Review by farshad68 — COMMENTED** [review]
two points come to my mind. I think we should add config to use this feature in the config file. the second one is the size of the queries, they are huge and while I try to run them I could not. I don't know it is possible to run these size queries or not (maybe use POST not GET).
here is one of the generated queries, [here](https://docs.google.com/document/d/1fumAtzjEjMxRPhj2s0nWr1_yJMq-A7TEJudvCLXv2WA/edit?usp=sharing) , they face 400 HTTP responses.

File: service/src/test/java/org/dice_research/fc/paths/scorer/count/PropPathBasedPairCountRetrieverTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.dice_research.fc.paths.PredicateFactory;
+import org.dice_research.fc.paths.scorer.ICountRetriever;
+import org.dice_research.fc.paths.scorer.count.max.DefaultMaxCounter;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class PropPathBasedPairCountRetrieverTest {
+
+  @Test
+  public void test() {
```
[the remark this answers is not in the document]
Comment (MichaelRoeder) [reply service/src/test/java/org/dice_research/fc/paths/scorer/count/PropPathBasedPairCountRetrieverTest.java:25]: I fully agree that the method names should be self-explaining. When I write `test` as method name, the method is the only test method in the class and typically directly tests the main functionality of the class. I am not sure how I should name them instead :confused:

**Comment by MichaelRoeder** [comment]
@TortugaAttack did you encounter HTTP 400 errors when using the sampling within glisten? Did you use GET or POST? :thinking:

File: service/src/test/java/org/dice_research/fc/paths/scorer/count/PropPathBasedPairCountRetrieverTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (farshad68) [reply service/src/test/java/org/dice_research/fc/paths/scorer/count/PropPathBasedPairCountRetrieverTest.java:25]: you right, when there is only one method in the class and that method tests the main functionality of the class, naming it is not necessary.

**Comment by TortugaAttack** [comment]
GET and not after commit the latest commit dd33da7.
Before that there were invalid queries using blank nodes as properties.

**Comment by MichaelRoeder** [comment]
@farshad68 The HTTP 400 problem should be fixed. Please check again. :slightly_smiling_face:

**Pull request closed without merging**